tapeagents/llms.py

```diff
@@ -525,9 +537,22 @@ def _generate(self, prompt: Prompt) -> Generator[LLMEvent, None, None]:
             data = r.json()
             try:
                 content = data["choices"][0]["message"]["content"]
+                if self.remove_leading_white_space:
+                    # vllm sometimes adds a whitespace at the beginning of the completion
+                    content = content.lstrip()
                 if not content:
                     logger.warning(f"Empty completion {data}")
-                output = LLMOutput(content=content)
+
+                if self.collect_logprobs:
+                    # Note: vLLM tokens do not always match the HF tokenizer tokens
```

Comment:
I think this comment is misleading. Tokenizing the content on TapeAgents side will produce the same tokens. The issue you are alluding to is that the content tokens will be different when the content is tokenized together with the prompt. But this warning will be more appropriate in a different part of the codebase.

Reply:
I moved it into `orchestrate_rl.py` as `# Note: tokens produced during generation are not always the same as the tokens produced on the full sequence`